src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -494,9 +494,9 @@ private boolean addConnection()
          String password = config.getPassword();
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
-         poolElf.setupConnection(connection, connectionTimeout);
+         PoolBagEntry pbe = poolElf.createBagEntry(connection, connectionTimeout, this);
 
-         connectionBag.add(new PoolBagEntry(connection, this));
+         connectionBag.add(pbe);
```

Comment:
What I mean is, on the left I clearly see a method ``setupConnection()``, which clearly indicates what it does.

With the ``createBagEntry()`` call, I have no indication that it might execute SQL or otherwise change the connection state.  It "appears" to simply create a ``PoolBagEntry``.

Lastly, passing ``connectionTimeout`` into ``createBagEntry()`` make it appear that it is a part of the bag entry *state*.  However, it is only used during connection setup, which is crystal clear in the code on the left.